test/macros.jl

```diff
@@ -176,6 +176,22 @@ function custom_expression_test(ModelType::Type{<:JuMP.AbstractModel})
     end
 end
 
+function JuMP.parse_one_operator_constraint(_error::Function, ::Bool, ::Val{:f}, x)
+    return :(), :(build_constraint($_error, $(esc(x)), $(esc(CustomType()))))
+end
+function custom_function_test(ModelType::Type{<:JuMP.AbstractModel})
+    @testset "Custom function" begin
+        model = ModelType()
+        @variable(model, x)
+        @constraint(model, con_ref, f(x))
+        con = JuMP.constraint_object(con_ref)
+        @test jump_function(con) == x
+        @test moi_set(con) isa CustomSet
+
+        @test_macro_throws LoadError @constraint(model, g(x))
```

Comment:
Replace `LoadError` by `ErrorException`. This is why travis is failing.

Reply:
I didn't expect the macro to be so smart! Done in the next commit.